classes/stageAssignment/StageAssignmentDAO.php

```diff
@@ -367,12 +371,12 @@ public function deleteByAll($submissionId, $userGroupId, $userId)
      * @param int $stageId optional
      * @param int $userGroupId optional
      * @param int $userId optional
-     * @param int $roleId optional ROLE_ID_...
+     * @param int[] $roleIds optional [ROLE_ID_...]
      * @param bool $single specify if only one stage assignment (default is a ResultFactory)
      *
      * @return StageAssignment|ResultFactory Mixed, depending on $single
      */
-    public function _getByIds($submissionId = null, $stageId = null, $userGroupId = null, $userId = null, $roleId = null, $single = false)
+    public function _getByIds($submissionId = null, $stageId = null, $userGroupId = null, $userId = null, $roleIds = null, $single = false)
```

Comment:
I would suggest adding a typehint for the `$roleIds` parameter in particular, in case any 3rd-party code is passing an `int` here and doesn't catch the change to `array`.

Reply:
The way that it's currently set up, the function will work if someone passes an `int` to the `$roleIds`. We coerce the value to an array, which should turn `1` into `[1]`:

```php
$sanitizedRoleIds = join(',', array_map(fn ($roleId) => (int) $roleId, (array) $roleIds));
```

It's not pretty, but again I was erring on the side of caution. Happy to add the type hint if you want.

Reply:
Ah -- that's fine, I missed that. For new code I prefer strict typing (it's so easy to just throw `[]` around a single value) but for deprecated code like this backwards compatibility is probably the lesser evil.